conterrors/error.go

```diff
@@ -0,0 +1,10 @@
+package conterrors
```

Comment:
Don't make a package for this: just define the errors where they are thrown and keep a helper function that handles the various types.